### DISCUSSION TITLE
Can you upload a file to supabase storage bucket via backend API?

### DISCUSSION THREAD

**Discussion by riefer02** [supabase/supabase#9201] [Questions]
Hello Supabase team,

I am looking for a solution to handle the upload of a file to a storage bucket on the back end. 

My project website is built using the Fresh framework (Deno). 🦕 

I recently ran into an issue with the supabase JS Client when trying to upload a file to the storage bucket.

```
error: Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'href')
  if (!url) url = window.location.href
```

This makes sense because the code is not executed on the user's browser and the file upload handler occurs on the server.

My initial thought was to simply use an API to upload the file to the bucket, however, there doesn't appear to be any other way to upload files to the supabase storage except manually or through a client SDK according to the documentation I have found.

Is it possible to upload files to a storage bucket other than the SDK or manually?

Looking forward to hearing from you and keep up the good work. 

[Project Repo](https://github.com/riefer02/fresh-deno-api)
[Supabase Client](https://github.com/riefer02/fresh-deno-api/blob/main/utils/supabase-client.ts)
[Route](https://github.com/riefer02/fresh-deno-api/blob/main/routes/user/profile.tsx)

**Comment by GaryAustin1** [comment]
On SDK or manually...  You can just use the REST API for storage.
https://supabase.github.io/storage-api/

**Reply by riefer02** [reply]
@GaryAustin1 I was exploring that documentation and the API URL is:

```
https://supabase.github.io
```

Would I need to change that to my database's URL?

I'm also not sure what to put for the `wildcard` query param value. 

Not sure if I need to attach the file to the request body either.

Thanks for any advice in advance. Appreciate the help.

**Reply by ilbertt** [reply]
@riefer02 maybe this page could also help you: https://supabase.com/docs/reference/storage/usage.

> I'm also not sure what to put for the `wildcard` query param value.

As you see from https://supabase.com/docs/reference/storage/usage#upload-a-new-object, `wildcard` is just the path+file name of the new object you upload

> Not sure if I need to attach the file to the request body either.

Yes, you just need to put the file Buffer in HTTP POST request body.

See also my discussion https://github.com/supabase/supabase/discussions/9211, in which I ask if there are better specifications also for other endpoints (`/auth`, ...)
